server/blueprints/buySellBlueprint.py

```diff
@@ -0,0 +1,91 @@
+from sqlalchemy import select
+import sqlalchemy
+from sqlalchemy.orm import sessionmaker, Session
+import flask
+from auth.auth_tools import login_required
+from database.schema import Company, t_companyhistory
+# from plotly.offline import plot
+# from plotly.graph_objs import Scatter
+
+
+def create_blueprint(Makesession: sessionmaker):
+    buySellBlueprint = flask.Blueprint("buySellBlueprint", __name__)
+
+    @buySellBlueprint.route('/', methods=['GET'])
+    @login_required(Makesession)
+    def getStockDetails():
+
+        response_body = {
+            "error": None,
+            "data": None
+        }
+
+        session: Session = Makesession()
+        companyID: str = flask.request.args.get('company')
+
+        if companyID == None or companyID == "":
+            response_body["error"] = "No company name passed"
+            res = flask.make_response(flask.jsonify(response_body), 400)
+            return res
+
+        companyID = companyID.upper()  # All company ids are in uppercase
+
+        companyHistoryQuery = session.query(t_companyhistory.columns.time_fetched, t_companyhistory.columns.trading_price).filter(
+            t_companyhistory.columns.company_id == companyID)
+
+        companyHistory = session.execute(companyHistoryQuery).all()
+
+        response = [x._asdict() for x in companyHistory]
+
+        companyDetailsQuery = session.query(
+            Company.company_name, Company.num_shares).filter(Company.company_id == companyID)
+
+        companyDetails = session.execute(companyDetailsQuery).one()
+
+        session.close()
+
+        # response: {time_fetched:int, trading_price:str}[]
+
+        response_body["data"] = {
+            "stockData": response,
+            "companyDetails": companyDetails._asdict()
+        }
+
+        res = flask.make_response(flask.jsonify(response_body), 200)
+        return res
+
+    @buySellBlueprint.route('/', methods=['POST'])
+    @login_required(Makesession)
+    def executeTransaction():
+        response_body = {
+            "error": None,
+            "ok": False
+        }
+
+        session: Session = Makesession()
```

Comment:
Same here, maybe extract to another function

Reply:
Where do you think I should put the function?

Reply:
That was a bit confusing I realize.The function it'self is good, I just noticed that the body of the func was a bit long and could be broken up into a package for some type of functionality

Reply:
That way you can read over the blueprint and it's really clear what it takes as input, what it does, and what it gives as output